control-plane/api-gateway/binding/validation.go

```diff
@@ -226,6 +227,32 @@ func validateTLS(gateway gwv1beta1.Gateway, tls *gwv1beta1.GatewayTLSConfig, res
 	return nil, refsErr
 }
 
+func validateJWT(gateway gwv1beta1.Gateway, listener gwv1beta1.Listener, resources *common.ResourceMap) error {
+	policy, _ := resources.GetPolicyForGatewayListener(gateway, listener)
+	if policy == nil {
+		return nil
+	}
+
+	if policy.Spec.Override != nil && policy.Spec.Override.JWT != nil {
+		for _, provider := range policy.Spec.Override.JWT.Providers {
+			_, ok := resources.GetJWTProviderForGatewayJWTProvider(provider)
+			if !ok {
+				return errListenerJWTProviderNotFound
```

Comment:
Here and below, you can add specific information about the name of the provider not found:

```suggestion
				return fmt.Errorf("%w, %s", errListenerJWTProviderNotFound, provider)
```

Reply:
You could combine this with returning multiple errors to return all issues with providers for both default and overrides in one go. That way if a user has issues in both sections, they can find and fix them all at once.

Basically, not returning early and instead returning a slice of errors.

Reply:
good call!

Reply:
actually for this we don't care about that, the gateway policy is going to be the one with all the information about what is wrong, for this all we care about is if any are incorrect